requests_cache/policy/expiration.py

```
@@ -26,6 +26,13 @@ def get_expiration_datetime(
     ignore_invalid_httpdate: bool = False,
 ) -> Optional[datetime]:
     """Convert an expiration value in any supported format to an absolute datetime"""
+    # Invalid dates MUST be treated as 'already expired' (RFC 2616, section 14.21)
+    # Expires headers arrive as integer strings (eg. '0'...or even '-1' if you are Azure...)
+    if (
+        isinstance(expire_after, str)
+        and (expire_after[1:] if expire_after[0] == '-' else expire_after).isdigit()
```

It looks like the condition `... else expire_after).isdigit()` will treat any integer string as invalid, right? I think this would be correct without the `else` condition.

---

Leaning into 'Expires' MUST be an [RFC 1123](https://datatracker.ietf.org/doc/html/rfc1123) date format and that invalid dates MUST be treated as 'already expired' (from RFC 2616, section 14.21).

If this was my codebase, I would be looking to *replace* the `raise ValueError(f'Invalid HTTP date: {expire_after}')` in this function with `return start_time or utcnow()`; no code here should ever throw, if it does not like something just log and ignore.

The reason I did not go down this path this is that one of you unit tests (`test_expiration.py:test_get_expiration_datetime__httpdate`) *wants* `ValueError` to be raised so I assume there is some history behind the decision here.

Instead I went with something more granular and targetted string integers which look to be not be under consideration in the codebase (whilst affecting my use case) so I added a clause to make all integer strings (positive and negative) invalid.

The PR is a proposal, one limited in missing the back story behind the logic here, ultimately it is up to you to decide the best approach knowing that history.

---

That's reasonable. I'll go ahead and merge this, and take another look to see if that `ValueError` is still necessary instead of log/ignore.

That condition is mainly meant for some particularly weird httpdate values that can show up in the `Last-Modified` header ([examples](https://gist.github.com/timvisee/fcda9bbdff88d45cc9061606b4b923ca)).